src/zenml/integrations/kubernetes/orchestrators/kube_utils.py

```diff
@@ -0,0 +1,394 @@
+
+from kubernetes import client as k8s_client
+from kubernetes import config as k8s_config
+from kubernetes.client.rest import ApiException
+
+from zenml.integrations.kubernetes.orchestrators.manifest_utils import (
+    build_cluster_role_binding_manifest_for_service_account,
+    build_mysql_deployment_manifest,
+    build_mysql_service_manifest,
+    build_namespace_manifest,
+    build_persistent_volume_claim_manifest,
+    build_persistent_volume_manifest,
+    build_service_account_manifest,
+)
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+class PodPhase(enum.Enum):
+    """Phase of the Kubernetes Pod.
+
+    Pod phases are defined in
+    https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase.
+    """
+
+    PENDING = "Pending"
+    RUNNING = "Running"
+    SUCCEEDED = "Succeeded"
+    FAILED = "Failed"
+    UNKNOWN = "Unknown"
+
+
+def is_inside_kubernetes() -> bool:
+    """Check whether we are inside a kubernetes cluster or on a remote host.
+
+    Returns:
+        True if inside a kubernetes cluster, else False.
+    """
+    try:
+        k8s_config.load_incluster_config()
+        return True
+    except k8s_config.ConfigException:
+        return False
+
+
+def load_kube_config(context: Optional[str] = None) -> None:
+    """Load the kubernetes client config.
+
+    Depending on the environment (whether it is inside the running kubernetes
+    cluster or remote host), different location will be searched for the config
+    file.
+
+    Args:
+        context: Name of the Kubernetes context. If not provided, uses the
+            currently active context.
+    """
+    try:
+        k8s_config.load_incluster_config()
+    except k8s_config.ConfigException:
+        k8s_config.load_kube_config(context=context)
+
+
+def sanitize_pod_name(pod_name: str) -> str:
+    """Sanitize pod names so they conform to kubernetes pod naming convention.
+
+    Args:
+        pod_name: Arbitrary input pod name.
+
+    Returns:
+        Sanitized pod name.
+    """
+    pod_name = re.sub(r"[^a-z0-9-]", "-", pod_name.lower())
+    pod_name = re.sub(r"^[-]+", "", pod_name)
+    return re.sub(r"[-]+", "-", pod_name)
+
+
+def pod_is_not_pending(pod: k8s_client.V1Pod) -> bool:
+    """Check if pod status is not 'Pending'.
+
+    Args:
+        pod: kubernetes pod.
+
+    Returns:
+        False if the pod status is 'Pending' else True.
+    """
+    return pod.status.phase != PodPhase.PENDING.value  # type: ignore[no-any-return]
+
+
+def pod_failed(pod: k8s_client.V1Pod) -> bool:
+    """Check if pod status is 'Failed'.
+
+    Args:
+        pod: kubernetes pod.
+
+    Returns:
+        True if pod status is 'Failed' else False.
+    """
+    return pod.status.phase == PodPhase.FAILED.value  # type: ignore[no-any-return]
+
+
+def pod_is_done(pod: k8s_client.V1Pod) -> bool:
+    """Check if pod status is 'Succeeded'.
+
+    Args:
+        pod: kubernetes pod.
```

Comment:
Just noticing that you sometimes 'kubernetes' and sometimes 'k8s' and sometimes 'Kubernetes'. I imagine these all refer to the same thing, but somehow in the back of my mind I wonder if that's the right assumption (i.e. perhaps the package/library vs the tool)? Perhaps nothing to do with this observation, however. I know we switch in other places in the codebase too.

Reply:
All changed to "Kubernetes".